src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/letter/InternalChangeDirectionDueDateLetterGenerator.java

```diff
@@ -0,0 +1,84 @@
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.Event;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.Callback;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.callback.PreSubmitCallbackStage;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.entities.ccd.field.Document;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.handlers.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.service.DocumentCreator;
+import uk.gov.hmcts.reform.iacasedocumentsapi.domain.service.DocumentHandler;
+
+@Component
+public class InternalChangeDirectionDueDateLetterGenerator implements PreSubmitCallbackHandler<AsylumCase> {
+
+    private final DocumentCreator<AsylumCase> documentCreator;
+    private final DocumentHandler documentHandler;
+
+    public InternalChangeDirectionDueDateLetterGenerator(
+        @Qualifier("internalChangeDueDateLetter") DocumentCreator<AsylumCase> documentCreator,
+        DocumentHandler documentHandler
+    ) {
+        this.documentCreator = documentCreator;
+        this.documentHandler = documentHandler;
+    }
+
+    public boolean canHandle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        Objects.requireNonNull(callbackStage, "callbackStage must not be null");
+        Objects.requireNonNull(callback, "callback must not be null");
+
+        AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+
+        Event event = callback.getEvent();
+
+        return callbackStage == PreSubmitCallbackStage.ABOUT_TO_SUBMIT
+               && Event.CHANGE_DIRECTION_DUE_DATE == event
+               && isInternalCase(asylumCase)
+               && isAppellantInDetention(asylumCase)
+               && isDirectionPartyAppellantOrAppellantRespondent(asylumCase);
+    }
+
+    public PreSubmitCallbackResponse<AsylumCase> handle(
+        PreSubmitCallbackStage callbackStage,
+        Callback<AsylumCase> callback
+    ) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final CaseDetails<AsylumCase> caseDetails = callback.getCaseDetails();
+        final AsylumCase asylumCase = caseDetails.getCaseData();
+
+        Document internalChangeDirectionDueDateLetter = documentCreator.create(caseDetails);
+
+        documentHandler.addWithMetadataWithoutReplacingExistingDocuments(
+            asylumCase,
+            internalChangeDirectionDueDateLetter,
+            NOTIFICATION_ATTACHMENT_DOCUMENTS,
+            DocumentTag.INTERNAL_CHANGE_DIRECTION_DUE_DATE_LETTER
+        );
+
+        return new PreSubmitCallbackResponse<>(asylumCase);
+    }
+
+    private boolean isDirectionPartyAppellantOrAppellantRespondent(AsylumCase asylumCase) {
```

Comment:
If this functionality will be needed in the future it may be worth taking this out to asylumCaseUtils?

Reply:
I will keep here for now, and if we see code being done multiple places, then move out!